highlighters/brackets/brackets-highlighter.zsh

```diff
@@ -59,11 +59,15 @@ _zsh_highlight_highlighter_brackets_paint()
         lastoflevel[$level]=$pos
         ;;
       [")]}"])
-        matchingpos=$lastoflevel[$level]
-        levelpos[$pos]=$((level--))
-	if _zsh_highlight_brackets_match $matchingpos $pos; then
-          matching[$matchingpos]=$pos
-          matching[$pos]=$matchingpos
+        if (( level > 0 )); then
+          matchingpos=$lastoflevel[$level]
+          levelpos[$pos]=$((level--))
+          if _zsh_highlight_brackets_match $matchingpos $pos; then
+            matching[$matchingpos]=$pos
+            matching[$pos]=$matchingpos
+          fi
+        else
+          levelpos[$pos]=-1
```

Comment:
Can you explain this -1?  Why couldn't this assignment be removed?  (And then I wouldn't have to start thinking whether the modulo later is going to do `(-1-1) % N + 1` and what works out to...)

Reply:
No real reason other than it's an invalid value. There needs to be an assignment so the error is highlighted (the loop is over `${(k)levelpos}`). The modulo branch won't be reached since `(( $+matching[$pos] ))` will always fail.

Reply:
+1